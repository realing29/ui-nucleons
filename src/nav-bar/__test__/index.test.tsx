import React from 'react';
import { mount } from 'enzyme';
import NavBar from '..';
import ArrowLeftSVG from '@dev-dep/ui-quarks/icons/24x24/Stroked/arrow-left';
import CrossSVG from '@dev-dep/ui-quarks/icons/24x24/Stroked/cross';
import CartSVG from '@dev-dep/ui-quarks/icons/24x24/Stroked/cart';
import PersonSVG from '@dev-dep/ui-quarks/icons/24x24/Stroked/person';

describe('<NavBar />', () => {
  it('should render just title and subtitle', () => {
    const wrapper = mount(
      <NavBar
        title='Hello'
        subtitle='World'
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle "bottomBordered" prop', () => {
    const wrapper = mount(
      <NavBar
        title='Hello'
        subtitle='World'
        bottomBordered
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render start link-button', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          start: { text: 'Foo' },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render start secondary link-button', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          startSecondary: { text: 'Foo' },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render end link-button', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          end: { text: 'Bar' },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render end secondary link-button', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          endSecondary: { text: 'Bar' },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render link-buttons', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          start: { text: 'Foo' },
          end: { text: 'Bar' },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render start icon-button', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          start: { icon: ArrowLeftSVG },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render end icon-button', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          end: { icon: ArrowLeftSVG },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render icon-buttons', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          start: { icon: ArrowLeftSVG },
          end: { icon: CrossSVG },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render four icon-buttons', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          start: { icon: ArrowLeftSVG },
          startSecondary: { icon: CartSVG },
          end: { icon: CrossSVG },
          endSecondary: { icon: PersonSVG },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render mixed buttons', () => {
    const wrapper = mount(
      <NavBar
        title='Очень очень очень длинный заголовок'
        subtitle='Очень очень очень очень длинный подзаголовок'
        buttons={{
          start: { text: 'Foo' },
          end: { icon: CrossSVG },
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
