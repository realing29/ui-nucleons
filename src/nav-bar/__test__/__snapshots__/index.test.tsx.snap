// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should handle "bottomBordered" prop 1`] = `
<NavBar
  bottomBordered={true}
  subtitle="World"
  title="Hello"
>
  <div
    className="nav-bar inner-border-bottom"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Hello
      </div>
      <div
        className="subtitle truncate"
      >
        World
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render end icon-button 1`] = `
<NavBar
  buttons={
    Object {
      "end": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon stub"
            type="button"
          />
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
        >
          <button
            aria-hidden={false}
            className="button icon"
            type="button"
          >
            <ForwardRef(ArrowLeftSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M11.707 3.293a1 1 0 010 1.414L4.414 12l7.293 7.293a1 1 0 01-1.414 1.414l-8-8a1 1 0 010-1.414l8-8a1 1 0 011.414 0z"
                  fillRule="evenodd"
                />
                <path
                  clipRule="evenodd"
                  d="M3 12a1 1 0 011-1h17a1 1 0 110 2H4a1 1 0 01-1-1z"
                  fillRule="evenodd"
                />
              </svg>
            </ForwardRef(ArrowLeftSVG)>
          </button>
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render end link-button 1`] = `
<NavBar
  buttons={
    Object {
      "end": Object {
        "text": "Bar",
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          stub={true}
          text="Bar"
        >
          <button
            aria-hidden={true}
            className="button stub"
            type="button"
          >
            Bar
          </button>
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          text="Bar"
        >
          <button
            aria-hidden={false}
            className="button"
            type="button"
          >
            Bar
          </button>
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render end secondary link-button 1`] = `
<NavBar
  buttons={
    Object {
      "endSecondary": Object {
        "text": "Bar",
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          stub={true}
          text="Bar"
        >
          <button
            aria-hidden={true}
            className="button stub"
            type="button"
          >
            Bar
          </button>
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          text="Bar"
        >
          <button
            aria-hidden={false}
            className="button"
            type="button"
          >
            Bar
          </button>
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render four icon-buttons 1`] = `
<NavBar
  buttons={
    Object {
      "end": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "endSecondary": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "start": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "startSecondary": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          shift="right"
        >
          <button
            aria-hidden={false}
            className="button icon shift-right"
            type="button"
          >
            <ForwardRef(ArrowLeftSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M11.707 3.293a1 1 0 010 1.414L4.414 12l7.293 7.293a1 1 0 01-1.414 1.414l-8-8a1 1 0 010-1.414l8-8a1 1 0 011.414 0z"
                  fillRule="evenodd"
                />
                <path
                  clipRule="evenodd"
                  d="M3 12a1 1 0 011-1h17a1 1 0 110 2H4a1 1 0 01-1-1z"
                  fillRule="evenodd"
                />
              </svg>
            </ForwardRef(ArrowLeftSVG)>
          </button>
        </NavButton>
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          shift="left"
        >
          <button
            aria-hidden={false}
            className="button icon shift-left"
            type="button"
          >
            <ForwardRef(CartSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M0 3a1 1 0 011-1h3a1 1 0 01.956.706L8.739 15h9.48l1.5-6H10a1 1 0 110-2h11a1 1 0 01.97 1.243l-2 8A1 1 0 0119 17H8a1 1 0 01-.956-.706L3.261 4H1a1 1 0 01-1-1z"
                  fillRule="evenodd"
                />
                <path
                  d="M12 21a2 2 0 11-4 0 2 2 0 014 0zM19 21a2 2 0 11-4 0 2 2 0 014 0z"
                />
              </svg>
            </ForwardRef(CartSVG)>
          </button>
        </NavButton>
      </div>
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          shift="right"
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon shift-right stub"
            type="button"
          />
        </NavButton>
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          shift="left"
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon shift-left stub"
            type="button"
          />
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          shift="right"
        >
          <button
            aria-hidden={false}
            className="button icon shift-right"
            type="button"
          >
            <ForwardRef(PersonSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  d="M6.936 14.149a1 1 0 01-.585 1.287c-1.69.634-2.512 1.774-2.93 2.765a5.66 5.66 0 00-.37 1.305 4.584 4.584 0 00-.05.482V20.009l.005.085c.007.08.02.19.048.315.055.248.142.451.248.578A1.088 1.088 0 003.47 21H20.53a1.088 1.088 0 00.168-.013c.106-.127.193-.33.248-.578a2.473 2.473 0 00.053-.4V19.988a4.575 4.575 0 00-.05-.481 5.66 5.66 0 00-.37-1.306c-.418-.99-1.24-2.13-2.93-2.765a1 1 0 01.702-1.872c2.31.866 3.488 2.476 4.07 3.86.289.685.432 1.313.504 1.773a6.581 6.581 0 01.073.725l.002.05V20L22 20l1 .001v.03a2.904 2.904 0 01-.013.236 4.49 4.49 0 01-.088.573c-.095.431-.306 1.074-.806 1.584-.51.518-1.181.576-1.563.576H3.47c-.382 0-1.053-.058-1.562-.576-.501-.51-.712-1.153-.807-1.584a4.474 4.474 0 01-.1-.809v-.019L1 20.005v-.003S1 20 2 20l-1-.001V19.972a3.002 3.002 0 01.01-.216 7.662 7.662 0 01.568-2.331c.583-1.385 1.761-2.995 4.07-3.861a1 1 0 011.288.585z"
                />
                <path
                  clipRule="evenodd"
                  d="M12 3C9.878 3 8 4.923 8 7.5S9.878 12 12 12s4-1.923 4-4.5S14.122 3 12 3zM6 7.5C6 4.002 8.599 1 12 1s6 3.002 6 6.5-2.599 6.5-6 6.5-6-3.002-6-6.5z"
                  fillRule="evenodd"
                />
              </svg>
            </ForwardRef(PersonSVG)>
          </button>
        </NavButton>
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          shift="left"
        >
          <button
            aria-hidden={false}
            className="button icon shift-left"
            type="button"
          >
            <ForwardRef(CrossSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M20.707 3.293a1 1 0 010 1.414l-16 16a1 1 0 01-1.414-1.414l16-16a1 1 0 011.414 0z"
                  fillRule="evenodd"
                />
                <path
                  clipRule="evenodd"
                  d="M3.293 3.293a1 1 0 011.414 0l16 16a1 1 0 01-1.414 1.414l-16-16a1 1 0 010-1.414z"
                  fillRule="evenodd"
                />
              </svg>
            </ForwardRef(CrossSVG)>
          </button>
        </NavButton>
      </div>
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          shift="right"
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon shift-right stub"
            type="button"
          />
        </NavButton>
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          shift="left"
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon shift-left stub"
            type="button"
          />
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render icon-buttons 1`] = `
<NavBar
  buttons={
    Object {
      "end": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "start": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
        >
          <button
            aria-hidden={false}
            className="button icon"
            type="button"
          >
            <ForwardRef(ArrowLeftSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M11.707 3.293a1 1 0 010 1.414L4.414 12l7.293 7.293a1 1 0 01-1.414 1.414l-8-8a1 1 0 010-1.414l8-8a1 1 0 011.414 0z"
                  fillRule="evenodd"
                />
                <path
                  clipRule="evenodd"
                  d="M3 12a1 1 0 011-1h17a1 1 0 110 2H4a1 1 0 01-1-1z"
                  fillRule="evenodd"
                />
              </svg>
            </ForwardRef(ArrowLeftSVG)>
          </button>
        </NavButton>
      </div>
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon stub"
            type="button"
          />
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
        >
          <button
            aria-hidden={false}
            className="button icon"
            type="button"
          >
            <ForwardRef(CrossSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M20.707 3.293a1 1 0 010 1.414l-16 16a1 1 0 01-1.414-1.414l16-16a1 1 0 011.414 0z"
                  fillRule="evenodd"
                />
                <path
                  clipRule="evenodd"
                  d="M3.293 3.293a1 1 0 011.414 0l16 16a1 1 0 01-1.414 1.414l-16-16a1 1 0 010-1.414z"
                  fillRule="evenodd"
                />
              </svg>
            </ForwardRef(CrossSVG)>
          </button>
        </NavButton>
      </div>
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon stub"
            type="button"
          />
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render just title and subtitle 1`] = `
<NavBar
  subtitle="World"
  title="Hello"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Hello
      </div>
      <div
        className="subtitle truncate"
      >
        World
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render link-buttons 1`] = `
<NavBar
  buttons={
    Object {
      "end": Object {
        "text": "Bar",
      },
      "start": Object {
        "text": "Foo",
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          text="Foo"
        >
          <button
            aria-hidden={false}
            className="button"
            type="button"
          >
            Foo
          </button>
        </NavButton>
      </div>
      <div
        className="group"
      >
        <NavButton
          stub={true}
          text="Bar"
        >
          <button
            aria-hidden={true}
            className="button stub"
            type="button"
          >
            Bar
          </button>
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          text="Bar"
        >
          <button
            aria-hidden={false}
            className="button"
            type="button"
          >
            Bar
          </button>
        </NavButton>
      </div>
      <div
        className="group"
      >
        <NavButton
          stub={true}
          text="Foo"
        >
          <button
            aria-hidden={true}
            className="button stub"
            type="button"
          >
            Foo
          </button>
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render mixed buttons 1`] = `
<NavBar
  buttons={
    Object {
      "end": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "start": Object {
        "text": "Foo",
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          text="Foo"
        >
          <button
            aria-hidden={false}
            className="button"
            type="button"
          >
            Foo
          </button>
        </NavButton>
      </div>
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon stub"
            type="button"
          />
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
        >
          <button
            aria-hidden={false}
            className="button icon"
            type="button"
          >
            <ForwardRef(CrossSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M20.707 3.293a1 1 0 010 1.414l-16 16a1 1 0 01-1.414-1.414l16-16a1 1 0 011.414 0z"
                  fillRule="evenodd"
                />
                <path
                  clipRule="evenodd"
                  d="M3.293 3.293a1 1 0 011.414 0l16 16a1 1 0 01-1.414 1.414l-16-16a1 1 0 010-1.414z"
                  fillRule="evenodd"
                />
              </svg>
            </ForwardRef(CrossSVG)>
          </button>
        </NavButton>
      </div>
      <div
        className="group"
      >
        <NavButton
          stub={true}
          text="Foo"
        >
          <button
            aria-hidden={true}
            className="button stub"
            type="button"
          >
            Foo
          </button>
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render start icon-button 1`] = `
<NavBar
  buttons={
    Object {
      "start": Object {
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
        >
          <button
            aria-hidden={false}
            className="button icon"
            type="button"
          >
            <ForwardRef(ArrowLeftSVG)
              aria-hidden="true"
              className="svg"
              fill="#212121"
            >
              <svg
                aria-hidden="true"
                className="svg"
                fill="#212121"
                height={24}
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M11.707 3.293a1 1 0 010 1.414L4.414 12l7.293 7.293a1 1 0 01-1.414 1.414l-8-8a1 1 0 010-1.414l8-8a1 1 0 011.414 0z"
                  fillRule="evenodd"
                />
                <path
                  clipRule="evenodd"
                  d="M3 12a1 1 0 011-1h17a1 1 0 110 2H4a1 1 0 01-1-1z"
                  fillRule="evenodd"
                />
              </svg>
            </ForwardRef(ArrowLeftSVG)>
          </button>
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          icon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          stub={true}
        >
          <button
            aria-hidden={true}
            className="button icon stub"
            type="button"
          />
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render start link-button 1`] = `
<NavBar
  buttons={
    Object {
      "start": Object {
        "text": "Foo",
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          text="Foo"
        >
          <button
            aria-hidden={false}
            className="button"
            type="button"
          >
            Foo
          </button>
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          stub={true}
          text="Foo"
        >
          <button
            aria-hidden={true}
            className="button stub"
            type="button"
          >
            Foo
          </button>
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;

exports[`<NavBar /> should render start secondary link-button 1`] = `
<NavBar
  buttons={
    Object {
      "startSecondary": Object {
        "text": "Foo",
      },
    }
  }
  subtitle="Очень очень очень очень длинный подзаголовок"
  title="Очень очень очень длинный заголовок"
>
  <div
    className="nav-bar"
    style={
      Object {
        "height": "56px",
      }
    }
  >
    <div
      className="button-groups"
    >
      <div
        className="group"
      >
        <NavButton
          text="Foo"
        >
          <button
            aria-hidden={false}
            className="button"
            type="button"
          >
            Foo
          </button>
        </NavButton>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="title truncate"
      >
        Очень очень очень длинный заголовок
      </div>
      <div
        className="subtitle truncate"
      >
        Очень очень очень очень длинный подзаголовок
      </div>
    </div>
    <div
      className="button-groups end"
    >
      <div
        className="group"
      >
        <NavButton
          stub={true}
          text="Foo"
        >
          <button
            aria-hidden={true}
            className="button stub"
            type="button"
          >
            Foo
          </button>
        </NavButton>
      </div>
    </div>
  </div>
</NavBar>
`;
